repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/contract/ContractPreProcessor.java

```diff
@@ -193,6 +193,16 @@ private void updateExistingVersion(EntityMutationContext context, AtlasEntity en
 
         context.addUpdated(entity.getGuid(), entity, entityType, vertex);
 
+        AtlasEntityComparator entityComparator = new AtlasEntityComparator(typeRegistry, entityRetriever, context.getGuidAssignments(), true, true);
+        AtlasEntityComparator.AtlasEntityDiffResult diffResult   = entityComparator.getDiffResult(entity, vertex, !storeDifferentialAudits);
+        RequestContext        reqContext           = RequestContext.get();
+        if (diffResult.hasDifference()) {
+            if (storeDifferentialAudits) {
+                diffResult.getDiffEntity().setGuid(entity.getGuid());
+                reqContext.cacheDifferentialEntity(diffResult.getDiffEntity());
+            }
+        }
+
```

Comment:
Exact same lines of code are already written in updateExistingVersion(context, entity, currentVersionEntity);

Given that datasetAttributeSync() will be called in all cases, I think you can remove this logic from updateExistingVersion completely

Reply:
datasetAttributeSync() sync information from the contract entity to the associated dataset entity. So in that function, this diff is to calculate the diff of the dataset, whereas in updateExistingVersion(), we are calculating the diff of the DataContract entity.

Reply:
Noted

Reply:
Move this logic to separate method then to avoid code duplication